src/Interpreters/ActionsDAG.h

```diff
@@ -292,38 +222,18 @@ class ActionsDAG
     /// Otherwise, return actions which inputs are from available_inputs.
     /// Returned actions add single column which may be used for filter.
     /// Also, replace some nodes of current inputs to constant 1 in case they are filtered.
-    ActionsDAGPtr splitActionsForFilter(const std::string & filter_name, bool can_remove_filter, const Names & available_inputs);
+    ActionsDAGPtr splitActionsForFilter(const std::string & filter_name, bool can_remove_filter, const Names & available_inputs, const ColumnsWithTypeAndName & all_inputs);
```

Comment:
Maybe give a different name with the previous function? They look completely different.

Reply:
Need detailed comment about `all_inputs`. It's completely not clear why we are passing it from outside.